main.c

```diff
@@ -143,20 +147,29 @@ bruteforce_rec(struct task_t *task,
         for (int i = 0; config->alphabet[i] != '\0'; ++i)
         {
             task->password[pos] = config->alphabet[i];
-            if (bruteforce_rec(task, config, pos + 1, context, handler))
+            if (bruteforce_rec_internal(task, config, context, handler, pos + 1))
                 return true;
         }
     }
     return false;
 } 
 
+bool
+bruteforce_rec(struct task_t *task,
+               struct config_t *config,
+               void *context,
+               password_handler_t handler)
+{
+    return bruteforce_rec_internal(task, config, context, handler, task->from);
+}
+
 bool
 bruteforce_iter(struct task_t *task,
                 struct config_t *config,
                 void *context,
                 password_handler_t handler)
 {
-    size_t size = strlen(config->alphabet) - 1;
+    size_t size = task->to - task->from;
```

Comment:
WTF?

Reply:
I already fixed it (and 3 issues below) in fourth commit (9804552).

Reply:
I already fixed it (and 3 issues below) in fourth commit (9804552).